onnx_caffe2/backend.py

```diff
@@ -297,7 +299,65 @@ def _create_gemm(cls, n, opset_version):
         return ops
 
     @classmethod
-    def _create_pad(cls, n, opset_version):
+    def _create_lstm(cls, onnx_model, model_helper, n, opset_version):
+        assert model_helper is not None, "cannot convert LSTMs through this codepath"
+        assert onnx_model is not None, "cannot convert LSTMs through this codepath"
+        allowed_attrs = {'hidden_size'}
+        for x in n.attrs.keys():
+            assert x in allowed_attrs, "unsupported attribute " + x + " to LSTM"
+
+        input_blob, W, R, B, sequence_lens, initial_h, initial_c = n.inputs
+
+        hidden_size = n.attrs['hidden_size']
+
+        # ad-hoc, informally-specified, bug-ridden, slow
```

Comment:
it's much more plausible to infer it from the weight matrix instead. It's much more likely to be a graph initializer and will break less often. In general, yes, we need proper shape inference

Reply:
++ I'll try switching to the weight matrix. Or maybe even both